04stable.autobuild

```diff
@@ -98,7 +98,11 @@ in_flavor 'master', 'stable' do
                 else Autoproj.shell_helpers?
                 end
 
-            if shell_helpers && shell = ENV['SHELL']
+            unless shell = ENV['SHELL']
+                Autoproj.warn("SHELL Variable not set, using fallback detection")
+                shell ||= %x[ps -p $$ -ocomm= | tr -d '\n']
```

Comment:
ENV['SHELL'] holds the full path, while `shell` now contains only the basename. So I would suggest to use `%x[ps -p $$ -ocomm= | xargs which].strip`

Reply:
I thought this doesn't pose a problem as the immediate next action on the variable is stripping it to its base name (see line 106). However I do agree that your proposed edit is cleaner. Going to edit this accordingly..